WordPress/Classes/ViewRelated/Stats/Shared Views/Stats Detail/SiteStatsDetailTableViewController.swift

```diff
@@ -73,14 +74,21 @@ class SiteStatsDetailTableViewController: UITableViewController, StoryboardLoada
     func configure(statSection: StatSection,
                    selectedDate: Date? = nil,
                    selectedPeriod: StatsPeriodUnit? = nil,
-                   postID: Int? = nil) {
+                   postID: Int? = nil,
+                   tableStyle: UITableView.Style = .grouped
```

Comment:
I'm not sure I'm a huge fan of directly passing a table view style in. I wonder if we should instead pass something like a `source` which could maybe be a `StatType` (insights or period)? The base VC can then make its own decisions about how to display the tableview based on that. What do you think?

Reply:
I agree with you that I'd prefer to not do that with styles and would prefer something else to better key on state.  However thinking about this more, would you mind if we leave this change for later?  I feel like we need a matrix of cases for styles (or at least I do) to handle all the conditions to make sure there is no break.

For example:
- followers has grouped style for first level insights
- followers has insetGrouped style for first level insights feature flagged
- followers has grouped style for 2nd level insights
- followers has insetGrouped style for 2nd level insights feature flagged
- followers has grouped style for first level period
- followers has grouped style for 2nd level period
- Additionally some of the new insight cards actually use periodStore data like views/visitors so the lines are feeling very blurry to me without fleshing out all the of above.

I'd like to approach this when all the new card types are converted and then I can fill the matrix and we can see the best way to create a state / enum to key on as I think our stateful 'key' so to speak will be a hybrid of states to check which style should be displayed.  How does this sound?